flang/test/Analysis/AliasAnalysis/alias-analysis-pack-array.fir

```diff
@@ -0,0 +1,43 @@
+// Test alias analysis queries for dummy arguments wired
+// through fir.pack_array.
+// fir.pack_array is a pass-through operation for FIR alias analysis.
+// RUN: fir-opt %s --test-fir-alias-analysis -split-input-file --mlir-disable-threading 2>&1 | FileCheck %s
+
+// The two pointers referencing two different maybe repacked
+// versions of the original dummy arguments do not alias:
+// CHECK-DAG: test1_y_repack(1)#0 <-> test1_x_repack(1)#0: NoAlias
+// CHECK-DAG: test1_x_orig(1)#0 <-> test1_y_orig(1)#0: NoAlias
+
+// Repacked dummy does not alias with another original dummy:
+// CHECK-DAG: test1_y_repack(1)#0 <-> test1_x_orig(1)#0: NoAlias
+// CHECK-DAG: test1_x_repack(1)#0 <-> test1_y_orig(1)#0: NoAlias
+
+// Repacked dummy may alias with its original:
+// CHECK-DAG: test1_x_repack(1)#0 <-> test1_x_orig(1)#0: MayAlias
+// CHECK-DAG: test1_y_repack(1)#0 <-> test1_y_orig(1)#0: MayAlias
+
+// Ideally, these should report MustAlias, but MayAlias
+// may work as well:
```

Comment:
Jean, as you suggested, I set `approximateSource` for `fir.pack_array`.  I guess it may be our best bet now.

These queries are the ones that look suboptimal, but it should probably not affect performance for the time being.

All, please let me know what you think about this change.  If we agree to keep it this way, I will update the doc.